data/org.freedesktop.diagnostic.portal.Desktop.xml

```diff
@@ -0,0 +1,125 @@
+ License as published by the Free Software Foundation; either
+ version 2.1 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library. If not, see <http://www.gnu.org/licenses/>.
+
+ Author: tytan652 <tytan652@tytanium.xyz>
+-->
+
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <!--
+    org.freedesktop.diagnostic.portal.Desktop:
+    @short_description: Portal diagnostic interface
+
+    This interface provides APIs related to desktop portals states.
+
+    This documentation describes version 1 of this interface.
+  -->
+  <interface name="org.freedesktop.diagnostic.portal.Desktop">
+
+    <!--
+      service-info:
+
+      Detail about the frontend service implementation (e.g. xdg-desktop-portal).
+
+      The property vardict contains the following properties:
+
+      * ``name`` (``s``)
+
+        Name of the service.
+
+      * ``version`` (``s``)
+
+        Version of the service.
+
+      * ``stable`` (``b``)
+
+        Indicate if the version of the service is unstable.
+        Assume unstable if ``name``, ``version`` or ``stable`` is not present
+    -->
+    <property name="service-info" type="a{sv}" access="read">
+      <annotation name="org.qtproject.QtDBus.QtTypeName" value="QVariantMap"/>
+    </property>
+
+    <!--
+      lockdown-impl:
+
+      Detail about the in use Lockdown portal implementation.
+
+      Implementation vardict contains the following properties:
+
+      * ``name`` (``s``)
+
+        Name of the implementation. If this properties is not present, no
+        implementation were found.
+
+      * ``version`` (``u``)
+
+        Version of the implementation if the interface is versioned.
+        Assume 1 if not present
+    -->
+    <property name="lockdown-impl" type="a{sv}" access="read">
+      <annotation name="org.qtproject.QtDBus.QtTypeName" value="QVariantMap"/>
+    </property>
+
+    <!--
+      access-impl:
+
+      Detail about the in use Access portal implementation.
+
+      Implementation vardict containing the same
+      as :ref:`org.freedesktop.diagnostic.portal.Desktop:lockdown-impl`.
+    -->
+    <property name="access-impl" type="a{sv}" access="read">
+      <annotation name="org.qtproject.QtDBus.QtTypeName" value="QVariantMap"/>
+    </property>
+
+    <!--
+      portals:
+
+      The list of portal with details about their state:
+
+      * ``name``` (``s``)
+
+        Short name of the portal, usually the last element of their interface
+        name.
+
+        Each portal can contain the following properties:
+
+          * ``implementation`` (``a{sv}``)
```

Comment:
Could just provide the name of the impl and let the caller correlate that with the impl data from another property.

Reply:
It's made that way to avoid having to update the interface for every new portal, unless it's an impl like Access or Lockdown.